spec/delayed/class_name_spec.rb

```diff
@@ -10,9 +10,18 @@
       end
     end
 
-    it 'assigns the class name of the payload object to class_name' do
-      Delayed::Job.enqueue TestJob.new
-      expect(job.class_name).to eq 'TestJob'
+    context 'when your job has a display_name method' do
+      it 'assigns the display name of the payload object to class_name' do
+        Delayed::Job.enqueue TestJobWithDisplayName.new
+        expect(job.class_name).to eq 'TestDisplayName'
+      end
+    end
```

Comment:
This is interesting, since it captures the behavior of `Foo.delay.bar` indirectly. I'm into it. 👍